src/types.jl

```diff
@@ -945,16 +945,16 @@ function ReplicaSystem(;
                         energy_units=u"kJ * mol^-1",
                         k=default_k(energy_units),
                         data=nothing)
-    D = AtomsBase.n_dimensions(boundary)
+    D = AtomsBase.n_dimensions(boundary[1])
```

Comment:
The other options can be passed either as 1 or n values, e.g. `pairwise_inters` and `replica_pairwise_inters`. Here you could have `boundary` and `replica_boundaries` with similar logic.